bundle/regal/ast.rego

```diff
@@ -136,6 +141,32 @@ find_vars(node) := [var |
 	some var in _find_vars(path, value, regal.last(path))
 ]
 
+# METADATA
+# description: |
+#  finds all vars declared in `rule` *before* the `location` provided
+#  note: this isn't 100% accurate, as it doesn't take into account `=`
+#  assignments / unification, but it's likely good enough since other rules
+#  recommend against those
+find_vars_in_local_scope(rule, location) := [var |
+	some var in find_vars(rule)
+	_before_location(var, location)
+]
+
+_before_location(var, location) if {
+	var.location.row < location.row
+}
+
+_before_location(var, location) if {
+	var.location.row == location.row
+	var.location.col < location.col
+}
+
+# METADATA
+# description: |
+#  similar to `find_vars_in_local_scope`, but returns all variable names in scope
+#  of the given location *and* the rule names present in the scope (i.e. module)
+find_names_in_scope(rule, location) := rule_names | {var.value | some var in find_vars_in_local_scope(rule, location)}
```

Comment:
oh that location for a set union took me a few seconds to understand. 😃

Reply:
Hehe! I made it a bit more explicit now by using temporary vars before the union.